.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,216 @@
+name: Pull Request CI
+run-name: ${{ github.actor }} is running Pull Request CI
+
+on:
+  pull_request_target:
+    types: [opened, labeled, reopened, synchronize]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    env:
+      AWS_BUCKET: code.s4d.io
+      AWS_REGION: us-east-1
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.13.1
+
+      - name: Clean yarn cache
+        run: yarn cache clean
+
+      - name: Initialize dependencies
+        run: yarn install
+
+      - name: Cache Dependencies
+        uses: actions/cache@v3
+        if: steps.validate-dependencies.outputs.cache-hit != 'true'
+        with:
+          path: "**/node_modules"
+          key: node-modules-${{ hashFiles('./yarn.lock') }}
```

Comment:
Are we on yarn already during this PR?

Reply:
We are not yet on yarn. Since this will go into a feature branch, and the very next PR that will go in is to convert the repo into a yarn workspace. Thats why to avoid doing the same task again, I have assumed that we are on yarn already